### ISSUE TITLE
Startup errors disappear in docker compose up, make it difficult to debug

### ISSUE THREAD

**Issue by pszabop** [opensearch-project/OpenSearch#5864]
**Describe the bug**
If a startup error such as insufficient memory occurs, the error is never shown, making it difficult to debug

for example, this error occurs using docker run but doesn't show up in docker-compose up

```
[Native memory allocation (mmap) failed to map 1073741824 bytes for committing reserved memory](https://forum.opensearch.org/t/native-memory-allocation-mmap-failed-to-map-1073741824-bytes-for-committing-reserved-memory/4258)
```

**To Reproduce**
1. Use the standard docker-compose.yml from the documentation
2. Use a machine with only 1GB of memory such as an AWS t2.micro
3. type `docker-compose up opensearch-node1` to start only one of the nodes
4. The process will exit with no useful error message
5. Compare to this command, which shows a useful error

`docker run -d -p 9200:9200 -p 9600:9600 -e "discovery.type=single-node" opensearchproject/opensearch:latest`

**Expected behavior**
The same errors should result, so that issues with `docker-compose.yml` can be debugged.

**Plugins**
none

**Screenshots**
see error messages noted above

**Host/Environment (please complete the following information):**
AWS t2.micro using recent wizard
Linux 5.10.157-139.675.amzn2.x86_64 https://github.com/opensearch-project/OpenSearch/pull/1 SMP Thu Dec 8 01:29:11 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
Docker version 20.10.17, build 100c701
Docker Compose version v2.15.1

**Additional context**
docker-compose install isn't done by yum on AWS linux, used instructions from [here](https://stackoverflow.com/questions/63708035/installing-docker-compose-on-amazon-ec2-linux-2-9kb-docker-compose-file)

**Comment by dblock** [comment]
I've run into this many times. I think showing the error in *RED* is probably helpful, but maybe there are better mechanisms to abort a `docker run` with an error? Ideas? I also wonder whether this issue belongs in opensearch-devops?

**Comment by saratvemulapalli** [comment]
@dblock yup looks like the errors the author is interested in are being wiped by docker but usually we do have all the logs in `opensearch.log`. Lets start with opensearch-build and if needed we could come back to opensearch.
